rmw_fastrtps_cpp/test/test_get_native_entities.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "osrf_testing_tools_cpp/scope_exit.hpp"
+
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+
+#include "rmw/error_handling.h"
+#include "rmw/rmw.h"
+
+#include "rmw_fastrtps_cpp/get_client.hpp"
+#include "rmw_fastrtps_cpp/get_participant.hpp"
+#include "rmw_fastrtps_cpp/get_publisher.hpp"
+#include "rmw_fastrtps_cpp/get_service.hpp"
+#include "rmw_fastrtps_cpp/get_subscriber.hpp"
+
+#include "test_msgs/msg/basic_types.h"
+#include "test_msgs/srv/basic_types.h"
+
+class TestNativeEntities : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    rmw_init_options_t options = rmw_get_zero_initialized_init_options();
+    rmw_ret_t ret = rmw_init_options_init(&options, rcutils_get_default_allocator());
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      rmw_ret_t ret = rmw_init_options_fini(&options);
+      EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    });
+    options.enclave = rcutils_strdup("/", rcutils_get_default_allocator());
```

Comment:
Should this memory be freed in the TearDown call? IDK if `rmw_init_options_fini` does it automatically

Reply:
It is supposed to (e.g. see [here](https://github.com/ros2/rmw_fastrtps/blob/47bb055921a91a216a175375e5198618ec082b7d/rmw_fastrtps_shared_cpp/src/rmw_init.cpp#L106)).